src/Java.Interop/.editorconfig

```diff
@@ -0,0 +1,149 @@
+# This files sets all rules from Gendarme as Code Analysis errors
```

Comment:
What is the "nature" of this file?  What causes errors to be generated?  *Any* build, meaning we'd see future failures in PR builds?  Or just builds within "some supported IDE"?

When an error is found, how is it reported?

What's the workflow?

Reply:
These are understood by `msbuild` and are passed to `csc.exe` which actually does the source code analysis at compile time (this is source code analysis, not binary analysis like Gendarme or FxCop):

```
csc.exe ... /analyzerconfig:C:\code\xamarin-android\external\Java.Interop\.editorconfig
```

They are reported by the compiler just like any other compiler generated warning or error, so this will now fail builds from the `msbuild` command line, in IDEs, and in CI/PR builds without any additional steps.

Example:
https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3261455&view=results